packages/shared/src/components/PostOptionsReadingHistoryMenu.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { ReactElement, useContext } from 'react';
+import { Item } from '@dailydotdev/react-contexify';
+import dynamic from 'next/dynamic';
+import { QueryClient, QueryKey, useQueryClient } from 'react-query';
+import { ReadHistoryPost } from '../graphql/posts';
+import ShareIcon from '../../icons/share.svg';
+import BookmarkIcon from '../../icons/bookmark.svg';
+import BookmarkIconFilled from '../../icons/filled/bookmark_filled.svg';
+import XIcon from '../../icons/x.svg';
+import useBookmarkPost from '../hooks/useBookmarkPost';
+import AuthContext from '../contexts/AuthContext';
+import { ReadHistoryInfiniteData } from '../hooks/useInfiniteReadingHistory';
+import { useCopyLink } from '../hooks/useCopyLink';
+
+const PortalMenu = dynamic(() => import('./fields/PortalMenu'), {
+  ssr: false,
+});
+
+export type PostOptionsReadingHistoryMenuProps = {
+  post: ReadHistoryPost;
+  onHidden?: () => unknown;
+  displayCopiedMessageFunc?: () => void;
+  onHide?: (postId: string) => unknown;
+};
+
+const MenuIcon = ({ Icon }) => {
+  return <Icon className="mr-2 text-2xl" />;
+};
+
+/* eslint-disable no-param-reassign */
+const updateReadingHistoryPost =
+  (
+    queryClient: QueryClient,
+    historyQueryKey: QueryKey,
+    update: (oldPost: ReadHistoryPost) => Partial<ReadHistoryPost>,
```

Comment:
Instead of a function, why not use a plain object here? I see you passed a node but was not necessarily used in the implementations.